### PR TITLE
lib: sms: Send SMS ACK asynchronously

### PR DESCRIPTION
**Pull request by trantanen**
AT+CNMA=1 has been sent with nrf_modem_at_printf synchronously. Changing it to use nrf_modem_at_cmd_async to avoid blocking system workqueue.
AT+CNMA=1 can take tens of seconds, if network is unresponsive, before modem's timers expire.

Jira: NCSDK-27510

### CODE REVIEW

File: lib/sms/sms.c [hunk 1]
```diff
@@ -79,16 +79,27 @@ AT_MONITOR_ISR(sms_at_handler_cmt, "+CMT", sms_at_cmd_handler_cmt, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cds, "+CDS", sms_at_cmd_handler_cds, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cms, "+CMS", sms_at_cmd_handler_cms, PAUSED);
 
+static void sms_ack_resp_handler(const char *resp)
+{
+	/* No need to do anything really */
+	if (strlen(resp) >= 2 && resp[0] == 'O' && resp[1] == 'K') {
```
Comment (tokangas) [inline lib/sms/sms.c:85]: The condition does not look correct.

File: lib/sms/sms.c [hunk 2]
```diff
@@ -79,16 +79,27 @@ AT_MONITOR_ISR(sms_at_handler_cmt, "+CMT", sms_at_cmd_handler_cmt, PAUSED);
...
  *
  * @param[in] work Unused k_work instance required for work handler signature.
  */
 static void sms_ack(struct k_work *work)
 {
-	int ret = nrf_modem_at_printf(AT_SMS_PDU_ACK);
+	int ret = nrf_modem_at_cmd_async(sms_ack_resp_handler, AT_SMS_PDU_ACK);
 
-	if (ret != 0) {
+	if (ret == -EINPROGRESS) {
```
Comment (tokangas) [inline lib/sms/sms.c:99]:
```suggestion
	if (ret == -NRF_EINPROGRESS) {
```

File: lib/sms/sms.c (hunk 1 above)
Comment (trantanen) [reply lib/sms/sms.c]: Oops. Added a condition for shorter than 2 characters also.

File: lib/sms/sms.c [hunk 3]
```diff
@@ -79,16 +80,28 @@ AT_MONITOR_ISR(sms_at_handler_cmt, "+CMT", sms_at_cmd_handler_cmt, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cds, "+CDS", sms_at_cmd_handler_cds, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cms, "+CMS", sms_at_cmd_handler_cms, PAUSED);
 
+static void sms_ack_resp_handler(const char *resp)
+{
+	/* No need to do anything really */
+	if (strlen(resp) < 2 ||
+	    (strlen(resp) >= 2 && resp[0] != 'O' && resp[1] != 'K')) {
```
Comment (tokangas) [inline lib/sms/sms.c:87]: The second `strlen()` is redundant.
```suggestion
	if (strlen(resp) < 2 || (resp[0] != 'O' && resp[1] != 'K')) {
```

**Review by tokangas — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Pull request merged**